### DISCUSSION TITLE
RavenDB 6.0.4 SinkTask got stopped and started several times because of Database record change

### DISCUSSION THREAD

**Discussion by GustavXie** [ravendb/ravendb#18135] [Q&A] [answered]
Hi, when we deploy RavenDB sink task on the cluster, we experienced Sink task stop and start several times because of Database record change. Also consumer created failed sometime. Could you help? Thanks

Below you can see the messages in RabbitMQ are not handled and got stuck: 
![image](https://github.com/ravendb/ravendb/assets/49243610/c6ee3dd0-6316-44a5-b894-baab86083b34)

We experienced start and stop several time:
![image](https://github.com/ravendb/ravendb/assets/49243610/d95d9c3b-7aac-49b7-a08f-b62dea3cbef3)

![image](https://github.com/ravendb/ravendb/assets/49243610/8271df60-82e2-4a51-90a4-f6a043abff4b)
![image](https://github.com/ravendb/ravendb/assets/49243610/dfe187e7-134a-4b3d-9a62-a14c8f1682d7)

**Comment by ppekrol** [comment]
Hi,

can you specify which version of RavenDB you are using? The last message that you pasted my indicate that there is a connectivity issue between a RavenDB node and RabbitMQ, maybe a firewall?

**Reply by GustavXie** [reply]
6.0.4 is the version I am using. The main issue is SinkTask got stopped and started several times because of Database record change, do you know any thing about it?

**Reply by ppekrol** [reply]
Is your cluster experiencing some kind of unstability? That could cause such an effect.
There is an option to pin the task to the node (the pin icon next to Responsible Node), that would force task to be used only on that node. Note: the task will not switch to another node in case of that node being down.

**Reply by GustavXie** [reply]
Thanks I am going to try it today, and I have another question about how could we monitor whether sink task connecting to RabbitMQ successfully or not. Do you have any API to check the status or any health check for sink task?

**Reply by garayx** [reply]
you can use `GetOngoingTaskInfoOperation` (https://ravendb.net/docs/article-page/6.0/csharp/server/ongoing-tasks/general-info#get-ongoing-task-info-operation) to see its `TaskConnectionStatus`

**Reply by GustavXie** [reply]
Btw I also see the logs said SinkTask is in the sleep mode and need to setup the wakeup time? So the consumer will be removed due to the sleep mode? Do you know anything on it? Thanks

**Reply by garayx** [reply]
can you paste the exact log line you are getting ?

**Comment by arekpalinski** [comment, marked as answer]
> Btw I also see the logs said SinkTask is in the sleep mode and need to setup the wakeup time? So the consumer will be removed due to the sleep mode? Do you know anything on it? Thanks

We'll be fixing that - https://issues.hibernatingrhinos.com/issue/RavenDB-22105/QueueSink-with-idle-database-will-stop-consuming
